microsoft/testsuites/power/power.py

```diff
@@ -0,0 +1,680 @@
+        )
+
+        log.info(
+            "Sent hibernate command to the VM and continue checking its status"
+            " in every 15 seconds until 10 minutes timeout.")
+
+        # Verify the VM status
+        # Can not find if VM hibernation completion
+        # or not as soon as it disconnects the network. Assume it is in timeout.
+        timeout = 600
+        start_time = time.time()
+        vm_stopped = False
+        while (time.time() - start_time < timeout):
+            time.sleep(15)
+            if not _is_vm_running(node, log, self.REBOOT_TIME_OUT):
+                vm_stopped = True
+                break
+            else:
+                log.debug("VM is not stopped. Waiting for 15s.......")
+
+        if vm_stopped:
+            log.info(
+                "Verified VM status is stopped after hibernation command sent")
+        else:
+            raise LisaException(
+                "Failed to change VM status to be stopped"
+                " after hibernation command sent")
+
+        # Resume the VM
+        start_stop.start()
+        log.info(
+            "Waked up the VM and continue checking its status"
+            " in every 15 seconds until 15 minutes timeout")
+
+        # Wait for VM resume for 15 min-timeout
+        timeout = 900
+        vm_resumed = False
+        start_time = time.time()
+        while (time.time() - start_time < timeout):
+            time.sleep(15)
+            is_ready, tcp_error_code = wait_tcp_port_ready(
+                node.public_address,
+                node.public_port,
+                log=log,
+                timeout=30,
+            )
+            if is_ready:
+                cmd_res = node.execute(
+                    "dmesg | grep -i 'hibernation exit'",
+                    shell=True,
+                    sudo=True
+                )
+                # TODO: THIS DMESG CHECKING APPROACH MAY NEED IMPROVE
+                res = re.search(r'(.*?hibernation exit.*?)',
+                                dmesg.get_output(force_run=True), re.IGNORECASE)
+                if not res or "hibernation exit" not in res.group(1):
+                    log.error(
+                        "VM resumed successfully "
+                        "but could not determine hibernation completion")
+                else:
+                    log.info("VM resumed successfully")
+                    vm_resumed = True
+                    break
+            else:
+                log.debug("VM is still resuming!")
+
+        if vm_resumed:
+            log.info("VM resume completed.")
+        else:
+            raise LisaException("VM resume did not finish.")
+
+        # Note that if you use NTP, the hardware clock is
+        # automatically synchronized to the system clock every 11 minutes,
+        # and this command is useful only at boot time
+        # to get a reasonable initial system time.
+        log.info("Waiting for RTC re-sync in 12 minutes...")
+        time.sleep(720)
```

Comment:
Don't hard code sleep in code. Try to use some signal. In this case, it should trigger to the time sync, instead of waiting it. Wait 10 minutes in each test case is terrible.

Reply:
change to a loop to check every 10 sec and timeout in 12mins

Reply:
How about 5 seconds? It saves up to 5 seconds more.